mlrun/feature_store/retrieval/dask_merger.py

```diff
@@ -50,10 +52,50 @@ def _generate_vector(
         # load dataframes
         feature_sets = []
         dfs = []
+        feature_sets_names = []
+        keys = []
+        all_columns = list()
+        self._parse_relations(feature_set_objects, feature_set_fields)
+
         for name, columns in feature_set_fields.items():
             feature_set = feature_set_objects[name]
             feature_sets.append(feature_set)
             column_names = [name for name, alias in columns]
+            feature_sets_names.append(name)
+
+            # updating the columns list that needed for tho join.
+            # Build left and right keys for the join apply
+            right_keys = []
+            left_keys = []
+            temp_key = []
+            for key, dict_relation in self._relation.items():
+                if re.findall(f":{name}$", key):
+                    for left, right in dict_relation.items():
+                        if (
+                            right not in column_names
+                            and right not in feature_set.spec.entities.keys()
+                        ):
+                            column_names.append(right)
+                            if self._drop_indexes:
+                                self._append_drop_column(right)
+                        if key.split(":")[0] in feature_sets_names:
+                            right_keys.append(right)
+                            left_keys.append(left)
+                        temp_key.append(right)
+                elif re.findall(f"^{name}:", key):
+                    for left, right in dict_relation.items():
+                        if (
+                            left not in column_names
+                            and left not in feature_set.spec.entities.keys()
+                        ):
+                            column_names.append(left)
+                            if self._drop_indexes:
+                                self._append_drop_column(left)
+                        if key.split(":")[1] in feature_sets_names:
+                            right_keys.append(left)
+                            left_keys.append(right)
```

Comment:
I didn't get why now its flipped. If the user gave a wrong order in the feature names he should be aware of that and it shouldn't work in my opinion.

(If I understand what you are doing here)